fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2208,9 +2212,33 @@ private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext se
             if (resource.getId() == null) {
                 throw new IllegalStateException("Returned resources must have an id.");
             }
+
+            // By default it's a MATCH, else if it's not a match it is of one of two types
+            // OUTCOME for operational outcomes embedded in the bundle
+            // INCLUDE for revinclude and include resources
+            SearchEntryMode mode = SearchEntryMode.MATCH;
+            String bundleEntryResourceType = resource.getClass().getSimpleName();
+            Search.Builder builder = Search.builder();
+            if (!resourceType.equals(bundleEntryResourceType)) {
```

Comment:
I believe it's possible for an 'included' resource to be of the same type as the matched resource, for example _include=Observation:derived-from or _include=Observation:has-member could return other Observation resources. Think this would mark it as MATCH instead of INCLUDE.

Reply:
unfortunately - we don't have enough details to differentiate.

Reply:
We'd have to modify the underlying jdbc persistence layer, and I don't think that's reasonable.

Reply:
Thoughts?

Reply:
I guess it comes down to whether an occasional inaccuracy in the typing is outweighed by the usefulness of having that typing. One other thought is that for include searches, maybe the count query be replaced with a query that actually returns the resource IDs, which you could then use to determine if the resource was a MATCH vs. INCLUDE. But I'm not familiar with the persistence layer yet, so don't know how big a deal that would be, and would probably be a slight performance hit as well.

Reply:
Yeah - it'd be a pretty big change.  We could document this in the conformance document?